### PR TITLE
Support non-uniform dimensionality in dependent variables

### PR DESCRIPTION
**Pull request by KirillZubov**

### CODE REVIEW

**Comment by KirillZubov** [comment]
```julia
@parameters t x
@variables u(..) v(..)

@derivatives Dt'~t
@derivatives Dx'~x

eqs = [
    Dt(u(t)) ~ Dx(v(x, t)),
    Dt(v(x, t)) ~ 1.0
]

bcs = [u(0.0) ~ 0.0,
       v(0.0, t) ~ 0.0,
       v(x, 0.0) ~ 0.0]

domains = [t ∈ IntervalDomain(0.0, 1.0),
           x ∈ IntervalDomain(0.0, 1.0)]

ind_vars = [t, x]
dep_vars = [u, v]

pde_system = PDESystem(eqs, bcs, domains, ind_vars, dep_vars);

dx = 0.1

chain1 = FastChain(FastDense(1,16,Flux.σ),FastDense(16,1))
chain2 = FastChain(FastDense(2,16,Flux.σ),FastDense(16,1))

u = get_u()
chains = [chain1,chain2]


stochastic_strategy = NeuralPDE.StochasticTraining(number_of_points = 100)
quadrature_strategy = NeuralPDE.QuadratureTraining(algorithm=HCubatureJL(),reltol= 1e-4,abstol= 1e-4,maxiters=1e2)
discretization = NeuralPDE.PhysicsInformedNN(chains, strategy=quadrature_strategy);
prob = NeuralPDE.discretize(pde_system, discretization)
pdeLoss, bcsloss = NeuralPDE.symbolic_discretize(pde_system, discretization)

res = GalacticOptim.solve(prob, ADAM(0.1); cb=cb, maxiters=500)
phi = discretization.phi

xs,ts = [domain.domain.lower:0.1:domain.domain.upper for domain in domains]

initθ = discretization.initθ
len_θ = length.(initθ)
sep =[ 1:len_θ[1], [len_θ[i]+1:len_θ[i]+len_θ[i+1] for i in 1:length(len_θ)-1]...]
minimizers = [res.minimizer[s] for s in sep]

u_predict1  = [phi[1]([t],minimizers[1])[1] for t in ts]
u_predict2  = [phi[2]([x,t],minimizers[2])[1] for x in xs  for t in ts]

p1 = plot(ts, u_predict1)
p2 = plot(xs, ts, u_predict2, st=:surface)
plot(p1,p2)
```
image

File: src/pinns_pde_solve.jl [hunk 1]
```diff
@@ -24,29 +22,37 @@ end
...
                            init_params = nothing;
                            _phi = nothing,
-                           autodiff=false,
                            _derivative = nothing,
                            strategy = GridTraining(),
                            kwargs...)
-    if init_params === nothing
-        initθ = DiffEqFlux.initial_params(chain)
+    if init_params == nothing
+        if chain isa Array
```
Comment (ChrisRackauckas) [inline src/pinns_pde_solve.jl:29]:
```suggestion
        if chain isa AbstractArray
```

File: src/pinns_pde_solve.jl [hunk 2]
```diff
@@ -24,29 +22,37 @@ end
...
+            initθ = DiffEqFlux.initial_params(chain)
+        end
+
     else
         initθ = init_params
     end
 
     if _phi == nothing
-        phi = get_phi(chain)
+        if chain isa Array
```
Comment (ChrisRackauckas) [inline src/pinns_pde_solve.jl:40]:
```suggestion
        if chain isa AbstractArray
```

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Looks like a test failure.

**Pull request merged**